xarray/core/formatting.py

```diff
@@ -253,8 +253,15 @@ def summarize_coord(name, var, col_width):
 
 
 def summarize_attr(key, value, col_width=None):
-    # ignore col_width for now to more clearly distinguish attributes
-    return u'    %s: %s' % (key, maybe_truncate(value))
+    """Summary for __repr__ - use ``X.attrs[key]`` for full value."""
+    key = u'    %s:' % unicode_type(key)
+    if col_width is not None:
+        key = pretty_print(key, col_width)
+    output = u'%s %s' % (key, unicode_type(value).replace(u'\t', u'    '))
```

Comment:
Indentation would certainly be an improvement, but I still strongly prefer to limit each attribute to a single line.

My motivation for this pull request is seeing a common pattern in notebooks dealing with NetCDF data of simply discarding the `attrs` metadata, because (a) the formatting is a mess and (b) it's a lot of output.  Taking the example in #1319, this is a temporary file with "only" 29 attributes, of which eight are long enough to truncate and three include newlines.  *This is a short example.*  With larger workflows, this pattern destroys provenance and reproducibility - along with directly useful information!

If users need the unmodified attributes or are unsure about types, they can access them directly - so I would really prefer to keep the repr to a single line for each, no matter what.